components/analytics/true_up_review.tsx

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useEffect} from 'react';
+import {FormattedMessage} from 'react-intl';
+
+import {useDispatch, useSelector} from 'react-redux';
+
+import moment from 'moment';
+
+import classNames from 'classnames';
+
+import {submitTrueUpReview, getTrueUpReviewStatus} from 'actions/hosted_customer';
+import {
+    isCurrentLicenseCloud,
+} from 'mattermost-redux/selectors/entities/cloud';
+
+import {
+    getSelfHostedErrors,
+    getTrueUpReviewProfile as trueUpReviewProfileSelector,
+    getTrueUpReviewStatus as trueUpReviewStatusSelector,
+} from 'mattermost-redux/selectors/entities/hosted_customer';
+
+import useCanSelfHostedSignup from 'components/common/hooks/useCanSelfHostedSignup';
+import CheckMarkSvg from 'components/widgets/icons/check_mark_icon';
+
+import './true_up_review.scss';
+import {GlobalState} from '@mattermost/types/store';
+import WarningIcon from 'components/widgets/icons/fa_warning_icon';
+import {isCurrentUserSystemAdmin} from 'mattermost-redux/selectors/entities/users';
+import {getLicense} from 'mattermost-redux/selectors/entities/general';
+import {TrueUpReviewProfileReducer, TrueUpReviewStatusReducer} from '@mattermost/types/hosted_customer';
+
+const TrueUpReview: React.FC = () => {
+    const dispatch = useDispatch();
+    const isCloud = useSelector(isCurrentLicenseCloud);
+    const isAirGapped = !useCanSelfHostedSignup();
```

Comment:
Didn't notice this before, but we'll need a different air-gapped check here. The self-hosted signup one is particular to the purchase flow in ways that make it unreliable here. It checks connectivity to Stripe by trying to load in stripe assets. This will indefinitely be able to be turned off by users who don't want this feature (see here https://github.com/mattermost/mattermost-server/blob/master/web/handlers.go#L239). It also checks whether signup is available in CWS, which we have the option to turn off this service if there's something wrong with it. We wouldn't want ability to do true up review turned off because the purchase flow is unavailable in CWS.

I think the simplest generalized check to add would be the one you originally implemented for self-hosted purchase, the one that does a GET on the `healthz` endpoint (https://github.com/mattermost/mattermost-server/pull/21693/files). We might want to call the endpoint something more general like `check-external-connectivity` or `check-cws-connection` though; `cws-health-check` makes it sound like we're interested in whether CWS is healthy as a service, which we don't care about per se, just whether we can connect to the wider internet).

I don't think we want to do true ICMP pings to a DNS; they often require special OS permissions or setups, or depend on shelling out and ping command being available, which might not be available on certain deployments.

Reply:
Gotcha, yeah my first thought was to use the one I originally implemented. I'll take your naming suggestions into account.